From 4ae219d804409c4056741c551a5b7d363d18de79 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Sun, 20 Oct 2024 22:07:22 +0200
Subject: [PATCH] rg35xx 2024: enable gpu

---
 .../sun50i-h700-anbernic-rg35xx-2024.dts      | 49 +++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
index 926367e52..4b2848310 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
@@ -175,6 +175,35 @@ reg_vcc5v: regulator-vcc5v { /* USB-C power input */
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 	};
+
+	gpu_opp_table: opp-table-1 {
+		compatible = "operating-points-v2";
+
+		opp-420000000 {
+			opp-hz = /bits/ 64 <420000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-456000000 {
+			opp-hz = /bits/ 64 <456000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-504000000 {
+			opp-hz = /bits/ 64 <504000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-552000000 {
+			opp-hz = /bits/ 64 <552000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-648000000 {
+			opp-hz = /bits/ 64 <648000000>;
+			opp-microvolt = <940000>;
+		};
+	};
 };
 
 &codec {
@@ -184,6 +213,26 @@ &codec {
 	status = "okay";
 };
 
+&gpu {
+	clocks = <&ccu /* CLK_PLL_GPU */ 8>, <&ccu CLK_GPU0>, <&ccu CLK_GPU1>, <&ccu CLK_BUS_GPU>;
+	clock-names = "clk_parent", "clk_mali", "clk_bak", "clk_bus";
+	gpu_idle = <1>;
+	dvfs_status = <1>;
+	operating-points = <0x9e340 0xea600 0x927c0 0xdbba0 0x86c40 0xdbba0 0x7b0c0 0xdbba0 0x6f540 0xdbba0 0x668a0 0xdbba0>;
+   #cooling-cells = <2>;
+	status = "okay";
+
+	ipa_dvfs {
+		compatible = "arm,mali-simple-power-model";
+		static-coefficient = <0x4268>;
+		dynamic-coefficient = <0x2ee>;
+		ts = <0x3e2da 0x2568 0xffffff98 0x04>;
+		thermal-zone = "gpu-thermal";
+		ss-coefficient = <0x24>;
+		ff-coefficient = <0x123>;
+	};	
+};
+
 &cpu0 {
 	cpu-supply = <&reg_dcdc1>;
 };
-- 
2.46.1

