From d5c1b26a0674d73e805011e4379682cd0b0b7249 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Sat, 27 Jul 2024 18:43:36 +0200
Subject: [PATCH 1/1] rg35xx 2024: add pwm backlight

---
 .../sun50i-h700-anbernic-rg35xx-2024.dts      | 30 +++++++++++--------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
index 24e302ca2b2a..5118a224f14a 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
@@ -22,9 +22,10 @@ aliases {
 	};
 
 	backlight: backlight {
-		compatible = "gpio-backlight";
-		gpios = <&pio 3 28 GPIO_ACTIVE_HIGH>; // PD28
-		default-on;
+		compatible = "pwm-backlight";
+		pwms = <&pwm 0 50000 0>;
+		pinctrl-0 = <&lcd_backlight_pin>;
+		pinctrl-names = "default";
 	};
 
 	battery: battery {
@@ -460,6 +451,19 @@ &usbphy {
 	status = "okay";
 };
 
+&pio {
+	lcd_backlight_pin: pwm0-pin {
+		pins = "PD28";
+		function = "pwm0";
+	};
+};
+
+&pwm {
+	allwinner,pwm-paired-channel-clock-sources = "hosc", "hosc", "hosc";
+	allwinner,pwm-paired-channel-clock-prescales = <0>, <0>, <0>;
+	status = "okay";
+};
+
 /* the AXP717 has USB type-C role switch functionality, not yet described by the binding */
 &usbotg {
 	dr_mode = "peripheral";   /* USB type-C receptable */
-- 
2.34.1

